File: app/src/main/java/org/michaelbel/moviemade/utils/LayoutHelper.java
package org.michaelbel.moviemade.utils;

import android.widget.FrameLayout;
import android.widget.LinearLayout;

import org.michaelbel.moviemade.Moviemade;

public class LayoutHelper {

    public static final int WRAP_CONTENT = -2;

    private static int getSize(float size) {
        return (int) (size < 0 ? size : DeviceUtil.INSTANCE.dp(Moviemade.AppContext, size));
    }

    public static FrameLayout.LayoutParams makeFrame(int width, int height, int gravity, float start, float top, float end, float bottom) {
        FrameLayout.LayoutParams params = new FrameLayout.LayoutParams(getSize(width), getSize(height));
        params.gravity = gravity;
        params.leftMargin = getSize(start);
        params.topMargin = getSize(top);
        params.rightMargin = getSize(end);
        params.bottomMargin = getSize(bottom);
        return params;
    }

    public static LinearLayout.LayoutParams makeLinear(int width, int height) {
        return new LinearLayout.LayoutParams(getSize(width), getSize(height));
    }
}

File: app/src/main/java/org/michaelbel/moviemade/utils/LayoutHelper.kt
package org.michaelbel.moviemade.utils

import android.widget.FrameLayout
import android.widget.LinearLayout

import org.michaelbel.moviemade.Moviemade

object LayoutHelper {

    const val WRAP_CONTENT = -2

    private fun getSize(size: Float): Int {
        return if (size < 0) {
            size.toInt()
        } else {
            DeviceUtil.dp(Moviemade.appContext, size)
        }
    }

    fun makeFrame(width: Int, height: Int, gravity: Int, start: Float, top: Float, end: Float, bottom: Float): FrameLayout.LayoutParams {
        val params = FrameLayout.LayoutParams(getSize(width.toFloat()), getSize(height.toFloat()))
        params.gravity = gravity
        params.leftMargin = getSize(start)
        params.topMargin = getSize(top)
        params.rightMargin = getSize(end)
        params.bottomMargin = getSize(bottom)
        return params
    }

    fun makeLinear(width: Int, height: Int): LinearLayout.LayoutParams {
        return LinearLayout.LayoutParams(getSize(width.toFloat()), getSize(height.toFloat()))
    }
}